### PR TITLE
About SRS

### PR DESCRIPTION
**Pull request by euggeze**
Wrote SRS-documentation and added link-pictures in SRS-file

### CODE REVIEW

File: documentation/SRS.md [hunk 1]
```diff
@@ -1,3 +1,185 @@
...
+### 1.1 Display list about employees
+The mode is designed to view list of employees in department.
+When user selects a department, a list of employees for that 
+department appears **(pic 1.1)**.
+
+#### *Main scenario*
+* User select item Employees;
+* Application open employees list.
+
+![list_employees](./search_period.png)
```
Comment (inauros) [inline documentation/SRS.md:29]: In accordance with the requirements, the average salary should be displayed for each department on the Departments page. I like the mockup above, so please don't change anything here. The proposed changes only need to be applied to the Departments page.

File: documentation/SRS.md [hunk 2]
```diff
@@ -1,3 +1,185 @@
...
+For dropping date and period lists user can click button **Drop**.
+
+### 1.3 Add employee
+
+#### *Main scenario*
+* User clicks button **Add**;
+* Application displays form for adding;
+* User enters information and presses **Save**;
+* If all information is valid, then information is adding in DB;
+* If don't all information is valid, then error message is displaying.
```
Comment (inauros) [inline documentation/SRS.md]: It would be great to see a mockup with an error message.

File: documentation/SRS.md [hunk 3]
```diff
@@ -1,3 +1,185 @@
...
+  then alert message is displaying.
+
+### 1.4 Edit employee
+
+#### *Main scenario*
+* User clicks button **Edit**;
+* Application displays form for editing;
+* User edits information and presses **Save**;
+* If all information is valid, then information is adding in DB;
+* If don't all information is valid, then error message is displaying.
```
Comment (inauros) [inline documentation/SRS.md]: It would be great to see a mockup with an error message.

File: documentation/SRS.md [hunk 4]
```diff
@@ -1,3 +1,185 @@
...
+**pic 2.1 - View the departments list**
+
+### 2.2 Add department
+
+#### *Main scenario*
+* User clicks button **Add**;
+* Application displays form for adding;
+* User enters information and presses **Save**;
+* If all information is valid, then information is adding in DB;
+* If don't all information is valid, then error message is displaying.
```
Comment (inauros) [inline documentation/SRS.md]: It would be great to see a mockup with an error message.

File: documentation/SRS.md [hunk 5]
```diff
@@ -1,3 +1,185 @@
...
+  then alert message is displaying.
+
+### 2.3 Edit department
+
+#### *Main scenario*
+* User clicks button **Edit**;
+* Application displays form for editing;
+* User edits information and presses **Save**;
+* If all information is valid, then information is adding in DB;
+* If don't all information is valid, then error message is displaying.
```
Comment (inauros) [inline documentation/SRS.md]: It would be great to see a mockup with an error message.

File: documentation/SRS.md [hunk 6]
```diff
@@ -1,3 +1,185 @@
...
+* User clicks "Yes", then information about employee
+will be deleted;
+* If user clicks "No", then window just will be closed and
+information won't be deleted.
+  
+#### *Cancel operation scenario*
+* User clicks button **Delete** in employee's line in lists;
+* An alert window will appear and ask if user want to delete;
+* User clicks "Yes", then information about employee
+will be deleted;
```
Comment (inauros) [inline documentation/SRS.md]: I find this scenario to be wrong or redundant.

File: documentation/SRS.md [hunk 7]
```diff
@@ -1,3 +1,185 @@
...
+* If user change information and don't save, 
+  then alert message is displaying.
+  
+### 1.5 Delete employee
+
+#### *Main scenario*
+* User clicks button **Delete** in employee's line in lists;
+* An alert window will appear and ask if user want to delete;
+* User clicks "Yes", then information about employee
+will be deleted;
```
Comment (inauros) [inline documentation/SRS.md]: What is expected behavior if an error occurs while deleteing?

File: documentation/SRS.md [hunk 8]
```diff
@@ -1,3 +1,185 @@
...
+![add_form](./add_employee.png)
+**pic 1.4 - View the form for adding**
+
+#### *Cancel operation scenario*
+* User clicks button **Add**;
+* Application displays form for adding;
+* User clicks button **Cancel**;
+* If user don't write information, then form will close;
+* If user writes information and don't save, 
+  then alert message is displaying.
```
Comment (inauros) [inline documentation/SRS.md]: It would be great to see a mockup for this case.

File: documentation/SRS.md [hunk 9]
```diff
@@ -1,3 +1,185 @@
...
+![add_form](./add_department.png)
+**pic 2.2 - View the form for adding**
+
+#### *Cancel operation scenario*
+* User clicks button **Add**;
+* Application displays form for adding;
+* User clicks button **Cancel**;
+* If user don't write information, then form will close;
+* If user writes information and don't save, 
+  then alert message is displaying.
```
Comment (inauros) [inline documentation/SRS.md:165]: It would be great to see a mockup for this case.

File: documentation/SRS.md [hunk 10]
```diff
@@ -1,3 +1,185 @@
...
+* User clicks "Yes", then information about department
+will be deleted;
+* If user clicks "No", then window just will be closed and
+information won't be deleted.
+  
+#### *Cancel operation scenario*
+* User clicks button **Delete** in department's line in lists;
+* An alert window will appear and ask if user want to delete;
+* User clicks "Yes", then information about department
+will be deleted;
```
Comment (inauros) [inline documentation/SRS.md]: I find this scenario to be wrong or redundant.

File: documentation/SRS.md [hunk 11]
```diff
@@ -1,3 +1,185 @@
...
+* If user change information and don't save, 
+  then alert message is displaying.
+  
+### 2.4 Delete department
+
+#### *Main scenario*
+* User clicks button **Delete** in department's line in lists;
+* An alert window will appear and ask if user want to delete;
+* User clicks "Yes", then information about department
+will be deleted;
```
Comment (inauros) [inline documentation/SRS.md:199]: 1. What is the expected behavior if an error occurs while deleteing?
2. What is the expected behavior if there is at least one employee associated with the department the user is trying to delete?

**Review by inauros — CHANGES_REQUESTED** [review]

File: documentation/SRS.md (hunk 1 above)
Comment (euggeze) [reply documentation/SRS.md:29]: Fixed

File: documentation/SRS.md (hunk 2 above)
Comment (euggeze) [reply documentation/SRS.md]: Done

File: documentation/SRS.md (hunk 3 above)
Comment (euggeze) [reply documentation/SRS.md]: Done

File: documentation/SRS.md (hunk 4 above)
Comment (euggeze) [reply documentation/SRS.md]: Done

File: documentation/SRS.md (hunk 5 above)
Comment (euggeze) [reply documentation/SRS.md]: Done

File: documentation/SRS.md (hunk 6 above)
Comment (euggeze) [reply documentation/SRS.md]: Fixed

File: documentation/SRS.md (hunk 7 above)
Comment (euggeze) [reply documentation/SRS.md]: If will be an error, an error message will appear (Added information in SRS)

File: documentation/SRS.md (hunk 8 above)
Comment (euggeze) [reply documentation/SRS.md:81]: Done

Comment (euggeze) [reply documentation/SRS.md]: Done

File: documentation/SRS.md (hunk 9 above)
Comment (euggeze) [reply documentation/SRS.md:165]: Done

File: documentation/SRS.md (hunk 10 above)
Comment (euggeze) [reply documentation/SRS.md]: Fixed

File: documentation/SRS.md (hunk 11 above)
Comment (euggeze) [reply documentation/SRS.md:199]: 1. If will be an error, an error message will appear (Added information in SRS)
2. All information about employees associated with the department will be deleted or I can suggest that they be recorded in a conditional "Temp" department

**Review by inauros — APPROVED** [review]

**Pull request merged**